src/sinks/datadog/metrics.rs

```diff
@@ -385,12 +436,59 @@ fn encode_events(events: Vec<Metric>, interval: i64, namespace: &str) -> Datadog
     DatadogRequest { series }
 }
 
+fn encode_distribution_events(
+    events: Vec<Metric>,
+    interval: i64,
+    namespace: &str,
+) -> DatadogRequest<DatadogDistributionMetric> {
+    debug!(message = "distribution", count = events.len());
+    let series = events
+        .into_iter()
+        .filter_map(|event| {
+            let fullname = encode_namespace(namespace, &event.name);
+            let ts = encode_timestamp(event.timestamp);
+            let tags = event.tags.clone().map(encode_tags);
+            match event.kind {
+                MetricKind::Incremental => match event.value {
+                    MetricValue::Distribution {
+                        values,
+                        sample_rates,
+                        statistic: StatisticKind::Summary,
+                    } => {
+                        let mut samples = Vec::new();
+                        for (v, c) in values.iter().zip(sample_rates.iter()) {
+                            for _ in 0..*c {
+                                samples.push(*v);
+                            }
+                        }
+
+                        if samples.is_empty() {
+                            None
+                        } else {
+                            Some(DatadogDistributionMetric {
+                                metric: fullname,
+                                interval: Some(interval),
+                                points: vec![DatadogPoint(ts, samples)],
+                                tags,
+                            })
+                        }
+                    }
+                    _ => None,
+                },
+                _ => None,
```

Comment:
Do these need to produce a warning about dropping the metric value?

Reply:
It is kinda blurry so I've gone with how the rest of the source deals with it. 

In the context of this source, if we ever get to first `None`, that is a logical bug in the source, while for the second `None` `Absolute` distribution doesn't make sense, and any other case is a logical error.

So maybe, we could just make sure we only pass `Incremental` `Summary` here, panic on anything else, and let the `encode_events` with nonsensical combinations as it has.

Or leave it like this as a slightly more generic/resistant function.